### PR TITLE
[Refactor] Init wallet parameters interaction

### PR DESCRIPTION
**Pull request by furszy**
This PR does two things:

1) Move the `pwalletMain` object inside the `wallet.h` file.
2) Move the wallet's parameters initialization to an static method `ParameterInteraction` in the wallet class.

This is coming (partially) from upstream 25340b7 .

### CODE REVIEW

**Comment by furszy** [comment]
You know, I thought on that but then thought on the translation context for the `AmountErrMsg` method, which will need to receive the parent or make the util file depend on `guiinterface` (which is not good, util.h shouldn't have any GUI dependency).

Good catch on the `InitError` and `InitWarning`. I missed them completely.

**Comment by random-zebra** [comment]
> which will need to receive the parent or make the util file depend on guiinterface

why? `AmountErrMsg` is only a string utility (takes a char pointer and a string to return a new string). It has no dependency with `guiinterface` (`UIError` and `UIWarning` do).

**Comment by furszy** [comment]
The `_` before the parenthesis is a translation function call which is part of `guiinterface`.

**Comment by random-zebra** [comment]
Right.
Well, couldn't this be directly inlined in guiinterface then?

**Comment by furszy** [comment]
it could but not really the file's responsibility. That file should be clean as possible from utility functions.
We should have `guiinterfaceutil` or well, need to check if `guiutil.h` is suited for this or not.

**Comment by random-zebra** [comment]
I don't think `guiutil.h` is a good place either. Would introduce unneeded dependencies in other files.
Probably a separate util file (in `src`, not in `src/qt`) would be better.

File: src/guiinterfaceutil.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2020 The PIVX developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef GUIINTERFACEUTIL_H
+#define GUIINTERFACEUTIL_H
+
+bool static UIError(const std::string &str)
```
Comment (random-zebra) [inline src/guiinterfaceutil.h]: why `static`? would leave it only if also `inline`d

**Comment by furszy** [comment]
Updated.

**Pull request merged**